react/week-3/learning-objectives.md

```diff
@@ -1,22 +1,11 @@
 ## Learning Objectives - React Week 3
 
-### Lifecycle
+### Data fetching (advanced)
 
-- "Clean up" a component when it is unmounted
-
-### Data fetching
-
-- Show some data from an API in a component
-  - Understand that `render` will be called before data is available and handle this appropriately
-  - Use `componentDidMount` lifecycle method to trigger `fetch()`
-  - Be able to hold data, loading & error states and render correctly based on these
+TODO (needs some scoping out)
```

Comment:
Could include:

- Update state when a prop changes (via prop watcher argument in useEffect hook)

This will most likely be linked to the form input point outlined below so not sure if it should be included in that section instead.

Reply:
Could you expand a bit on what you mean? Perhaps a quick code snippet? (Sorry I think my inexperience with Hooks is not helping here)

I was planning on including something explaining the 2nd dependencies argument to `useEffect()` in more detail. Basically something along the lines of [this section in the official docs](https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects).

And I was also planning on adding something about unsubscribing in `useEffect()`.

Reply:
I've pushed my initial ideas to the branch now